### ISSUE TITLE
Crash in BugsnagEvent.m line 303

### ISSUE THREAD

**Issue by sethfri** [bugsnag/bugsnag-cocoa#887]
### Describe the bug
Crash of unknown origin coming from Bugsnag. We have seen 5 crashes coming from 5 users so far.

### Steps to reproduce
Unknown

### Environment
* Bugsnag version: 6.2.1
* CocoaPods version: 1.9.3
* Carthage version (if any): n/a
* iOS/tvOS/macOS version(s): 14.0.0
* Simulator or physical device: iPhone XS Max
* XCode version: 12.1
* Swift version (if applicable): 5.1

### Stack Trace

<details><summary>Stack trace:</summary>

```
Crashed: InvitableContactManager_Typeahead_Queue
SIGSEGV 0x0000000000000020
0  libobjc.A.dylib                0x19dcfe160 objc_release + 16
1  CoreFoundation                 0x189b49304 -[__NSArrayI dealloc] + 80
2  MyApp                           0x104deb4a4 -[BugsnagEvent .cxx_destruct] + 303 (BugsnagEvent.m:303)
3  libobjc.A.dylib                0x19dcdf324 object_cxxDestructFromClass(objc_object*, objc_class*) + 112
4  libobjc.A.dylib                0x19dcf5d38 objc_destructInstance + 80
5  libobjc.A.dylib                0x19dcfc99c _objc_rootDealloc + 68
6  MyApp                           0x104ddd4f0 -[BugsnagClient notify:handledState:block:] + 1064 (BugsnagClient.m:1064)
7  MyApp                           0x104ddc708 -[BugsnagClient notifyError:block:] + 934 (BugsnagClient.m:934)
8  MyApp                           0x104dd4750 +[Bugsnag notifyError:block:] + 147 (Bugsnag.m:147)
9  MyApp                           0x1058cd2f8 specialized BugsnagLogStoreWriter.write(level:message:parameters:) + 4329689848 (<compiler-generated>:4329689848)
10 MyApp                           0x1058bede0 AggregateLogStoreWriter.write(level:message:parameters:) + 4329631200 (<compiler-generated>:4329631200)
11 MyApp                           0x1058beea0 protocol witness for LogStoreWriter.write(level:message:parameters:) in conformance AggregateLogStoreWriter + 4329631392 (<compiler-generated>:4329631392)
12 MyApp                           0x1058c5558 Logger.log(level:_:parameters:) + 4329657688 (<compiler-generated>:4329657688)
13 MyApp                           0x10579bc1c closure #1 in Type1.personAutocompletionItems(forQuery:) + 4328438812 (<compiler-generated>:4328438812)
14 MyApp                           0x105a9cf10 Type2.uniqueAutocompletionItemsForSearchTerms(in:passingTest:) + 318 (CCPersonManager.swift:318)
15 MyApp                           0x1048140b4 protocol witness for AutocompleteResultsProvider.uniqueAutocompletionItemsForSearchTerms(in:passingTest:) in conformance Type2 + 4312154292 (<compiler-generated>:4312154292)
16 MyApp                           0x10579b0d8 closure #1 in closure #1 in Type1.queryTypeaheadResults(withQuery:completionHandler:) + 305 (Type1.swift:305)
17 MyApp                           0x1046b5828 thunk for @escaping @callee_guaranteed () -> () + 4310718504 (<compiler-generated>:4310718504)
18 libdispatch.dylib              0x189862fd0 _dispatch_call_block_and_release + 32
19 libdispatch.dylib              0x189864ac8 _dispatch_client_callout + 20
20 libdispatch.dylib              0x18986bc08 _dispatch_lane_serial_drain + 580
21 libdispatch.dylib              0x18986c734 _dispatch_lane_invoke + 408
22 libdispatch.dylib              0x189876528 _dispatch_workloop_worker_thread + 708
23 libsystem_pthread.dylib        0x1d11b8908 _pthread_wqthread + 276
24 libsystem_pthread.dylib        0x1d11bf77c start_wqthread + 8
```
</details>

**Comment by johnkiely1** [comment]
Hi @sethfri,

Thanks for raising this, we will take a look at get back to you when we know more.

**Comment by mattdyoung** [comment]
This looks like it could be a bug in bugsnag-cocoa so we plan to investigate further.

**Comment by sethfri** [comment]
Edited the description to indicate that we're actually seeing this on Bugsnag 6.2.1, not 6.2.3. We haven't seen this crash yet with 6.2.3.

**Comment by sethfri** [comment]
Hi there, any update on this one? We've now seen 5 crashes coming from 5 users and don't appear to be seeing it at all in Bugsnag versions beyond 6.2.1. Would still be good to understand if the root cause was definitively fixed

**Comment by nickdowell** [comment]
Hi @sethfri we investigated this but cannot not positively identify the cause.

The crash has occurred when the Objective-C runtime was `dealloc`ing a `BugsnagEvent` and more specifically one of its `NSArray` properties.

There are several array properties on a `BugsnagEvent` and the stack trace unfortunately cannot identify which was involved, but I suspect that it was the `breadcrumbs` array. This would explain why you have not seen recurrences in more recent versions, after the fixes we made in v6.2.3
